spec/models/miq_expression_spec.rb

```diff
@@ -1631,4 +1631,146 @@
+        expression = {"INCLUDES" => {"field" => field, "value" => "abc"}}
+        expect(described_class.new(expression).sql_supports_atom?(expression)).to eq(true)
+      end
+
+      it "returns false for model.association.virtualfield" do
+        # delegation to #field_in_sql?
+        field = "ManageIQ::Providers::InfraManager::Vm.ext_management_system-hostname"
+        expression = {"INCLUDES" => {"field" => field, "value" => "abc"}}
+        expect(described_class.new(expression).sql_supports_atom?(expression)).to eq(false)
+      end
+
+      it "returns true for model.accociation.field" do
+        # delegation to #field_in_sql?
+        field = "ManageIQ::Providers::InfraManager::Vm.ext_management_system-name"
+        expression = {"INCLUDES" => {"field" => field, "value" => "abc"}}
+        expect(described_class.new(expression).sql_supports_atom?(expression)).to eq(true)
+      end
+
+      it "returns false if format of field is /\w+(\.\w+)+/ - interpreted as model.association..association-field" do
+        field = "ManageIQ::Providers::InfraManager::Vm.service.miq_request-v_approved_by"
+        expression = {"INCLUDES" => {"field" => field, "value" => "abc"}}
+        expect(described_class.new(expression).sql_supports_atom?(expression)).to eq(false)
+      end
+    end
+
+    context "not supported expression keys" do
+      it "returns false if expression key is 'FIND'" do
+        expect(described_class.new(nil).sql_supports_atom?("FIND" => {})).to eq(false)
+      end
+
+      it "returns false if expression key is 'REGULAR EXPRESSION MATCHES'" do
+        # field = ManageIQ::Providers::InfraManager::Vm-name
+        # expression = {"REGULAR EXPRESSION MATCHES" => {"filed" => field, "value" => "\w+"}}
+        expect(described_class.new(nil).sql_supports_atom?("REGULAR EXPRESSION MATCHES" => {})).to eq(false)
+      end
+
+      it "returns false if expression key is 'REGULAR EXPRESSION DOES NOT MATCH'" do
+        # field = ManageIQ::Providers::InfraManager::Vm-name
+        # expression = {"REGULAR EXPRESSION DOES NOT MATCH" => {"filed" => field, "value" => "\w+"}}
+        expect(described_class.new(nil).sql_supports_atom?("REGULAR EXPRESSION DOES NOT MATCH" => {})).to eq(false)
+      end
+
+      it "returns false if expression key is not 'CONTAINS' and operand is 'TAG'" do
+        # UI does not allow to create this kind of expression:
+        expression = {"=" => {"tag" => "Vm-vendor"}}
+        expect(described_class.new(expression).sql_supports_atom?(expression)).to eq(false)
+      end
+
+      it "returns false if operand is'COUNT' on model.association" do
+        association = "ManageIQ::Providers::InfraManager::Vm.users"
+        expression = {">" => {"count" => association, "value" => "10"}}
+        expect(described_class.new(expression).sql_supports_atom?(expression)).to eq(false)
+      end
+    end
+
+    context "delegated to #field_in_sql if expression key is not (CONTAINS || INCLUDE)" do
```

Comment:
i was not sure how to group 'regular' cases when all special cases/limitation in implementation passed. Is it better to  rename description to  
context "delegated to #field_in_sql", which still tie to implementation 
OR
 remove this context grouping and put directly under
 describe "#sql_supports_atom" ?